test/plugin_functional/plugins/core_plugin_a/public/application.tsx

```diff
@@ -115,8 +116,8 @@ const Nav = withRouter(({ history, navigateToApp }: NavProps) => (
   />
 ));
 
-const FooApp = ({ basename, context }: { basename: string; context: AppMountContext }) => (
-  <Router basename={basename}>
+const FooApp = ({ history, context }: { history: History; context: AppMountContext }) => (
+  <Router history={history}>
```

Comment:
Do we rely on the plugin tests to verify the `history` API functionality?

Reply:
The behavior that was outlined in the original bug is both verified in the FTR tests against this plugin and the jest integration tests in `router.test.tsx`